### DISCUSSION TITLE
Ecoleta em Deno - Primeira api com deno

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [frontendbr/forum#1817] [Discussão]
Fala galera! Tudo bem?

Acho que quase todo dev ouviu falar que semana passada rolou o e-coleta, além de fazer o que estava na semana fiz também em deno, estou finalizando o endpoint dos pontos com oak. 
Bom, apenas compartilhando para quem quiser brincar e fazer experiencia com deno também...

Quem achar legal da uma olhada lá no repo na pasta "server-deno":
[https://github.com/wilsonneto-dev/NextLevelWeek_EColeta](https://github.com/wilsonneto-dev/NextLevelWeek_EColeta)

Opiniões e PRs são super bem vindos!
Abraço a todos!

**Comment by wilsonneto-swe** [comment]
Apenas compartilhando, e também gostaria de opiniões sobre o que pensam a respeito do Deno. Eu vejo como uma ferramenta, acredito que vai coexistir com o node, mas que o node não vai perder força por ele. No fim apenas runtimes/ferramentas para rodar JS/ES/TS.

**Comment by suissa** [comment]
Mano pq vc precisa desse bind ??

```js

  .get("/items", itemsController.list.bind(itemsController))

```


Eu fiz uma brincadeira tb esses tempos com Deno achei legal mas falta mtooo ainda pra ser usado em prod.

**Comment by suissa** [comment]
https://github.com/suissa/deno-api-example

Ah na verdade eu peguei um repo q tinha um exemplo mto simples de API e refatorei ele Generalizando e Automatizando algumas coisas, tipo as rotas de cada módulo eu tenho um módulo genérico q cria elas:

```js
  
import { Router } from "https://deno.land/x/oak/mod.ts"

import genericRouter from "./../../generic.router.ts"
import ROUTES from "./routes.config.ts"

const to = (Router:any) => (router: any, route: any) => 
  genericRouter[route.method](Router, route)

const router = ROUTES.reduce(to(new Router()), {})

export default {
  router
}
```

Usando uma config:

```js
import Controller from "./controller.ts"

export default [
  {
    method: 'get',
    path: '/users',
    fn: Controller.find
  },
  {
    method: 'get',
    path: '/users/:id',
    fn: Controller.findOne
  },
  {
    method: 'post',
    path: '/users',
    fn: Controller.create
  },
  {
    method: 'put',
    path: '/users/:id',
    fn: Controller.update
  },
  {
    method: 'delete',
    path: '/users/:id',
    fn: Controller.remove
  }
]

```

**Comment by wilsonneto-swe** [comment]
@suissa esse bind é o que faço em todo método que é do controller, e apenas para que eu consiga usar o this lá dentro do método como sendo o controller, se eu não der o bind ele não interpreta o this lá dentro.
